src/librustc_passes/ast_validation.rs

```diff
@@ -318,6 +310,13 @@ impl<'a> Visitor<'a> for AstValidator<'a> {
 
     fn visit_generics(&mut self, g: &'a Generics) {
         let mut seen_default = None;
+        for lt_def in &g.lifetimes {
+            if lt_def.lifetime.ident.name == "'_" {
+                self.err_handler()
+                    .span_err(lt_def.lifetime.span,
+                              &format!("invalid lifetime name `{}`", lt_def.lifetime.ident));
+            }
```

Comment:
Feel free to bikeshed this error message (it shows up in code that tries to explicitly bind `'_` in the generics of an item declaration).

Reply:
An equivalent error is already issued for `'static`, so the wording can be reused:
```
invalid lifetime parameter name: `'static`
```